chainer/function_node.py

```diff
@@ -21,10 +21,13 @@
 import chainerx
 
 
-def _to_variable_with_chainerx_fallback_array(chainerx_array, fallback_array):
+def _to_variable_with_chainerx_fallback_array(
+        chainerx_device, chainerx_array, fallback_array):
     # chainerx_array can be None.
-    var = variable.Variable(
+    assert chainerx_array.device == chainerx_device.device
```

Comment:
`chainerx_array ` can be `None` (as actually already commented) in which case this assertion fails. Could you condition the assertion or take that into account?

Reply:
Thank you for pointing out. Fixed the condition.